py/asmrv32.c

```diff
@@ -326,10 +326,11 @@ void asm_rv32_emit_call_ind(asm_rv32_t *state, mp_uint_t index) {
 }
 
 void asm_rv32_emit_jump_if_reg_eq(asm_rv32_t *state, mp_uint_t rs1, mp_uint_t rs2, mp_uint_t label) {
-    ptrdiff_t displacement = (ptrdiff_t)(state->base.label_offsets[label] - state->base.code_offset);
+    mp_uint_t label_offset = state->base.label_offsets[label];
+    ptrdiff_t displacement = (ptrdiff_t)(label_offset - state->base.code_offset);
+    bool emit_short_jump = (label_offset != (mp_uint_t)-1) && (displacement < 0);
```

Comment:
I suggest making a local (static inline?) function that does these three lines of computation, and reused, to reduce code duplication.

Reply:
Good point!  I'll update my local copy and push this change along with fixes for the other failing tests, thanks.